src/librustc/mir/mod.rs

```diff
@@ -471,6 +464,7 @@ pub enum TerminatorKind<'tcx> {
 
         /// Possible values. The locations to branch to in each case
         /// are found in the corresponding indices from the `targets` vector.
+        // FIXME: ConstVal doesn’t quite make any sense here? Its a Switch*Int*.
         values: Vec<ConstVal>,
 
         /// Possible branch sites. The length of this vector should be
```

Comment:
This comment is not out of date. That being said I think it could make sense to adjust `SwitchInt` to be something like this instead:

```rust
    values: Vec<(ConstVal, BasicBlock)>,
    otherwise: Option<BasicBlock> // if None, is exhaustive Switch.
```

At the very least setup as above would make invariants obviously clear, rather than fuzzy like it is now. Does that sound sensible?